pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -493,15 +491,27 @@ func (nc *Controller) doNoScheduleTaintingPassWorker(i int, stopCh <-chan struct
 		select {
 		case <-stopCh:
 			return
-		case node := <-nc.nodeUpdateChannels[i]:
-			if err := nc.doNoScheduleTaintingPass(node); err != nil {
-				glog.Errorf("Failed to taint NoSchedule on node <%s>: %v", node.Name, err)
+		case nodeName := <-nc.nodeUpdateChannels[i]:
+			if err := nc.doNoScheduleTaintingPass(nodeName); err != nil {
+				glog.Errorf("Failed to taint NoSchedule on node <%s>: %v", nodeName, err)
 			}
 		}
 	}
 }
 
-func (nc *Controller) doNoScheduleTaintingPass(node *v1.Node) error {
+func (nc *Controller) doNoScheduleTaintingPass(nodeName string) error {
+	defer nc.nodeUpdateQueue.Done(nodeName)
```

Comment:
I moved `Done` to here to avoid requeue item during the process. The item (if any update during process) will requeue in `Done`.

Reply:
splitting Get and Done across methods makes this really hard to follow (even this change doesn't properly call done in stopCh cases).

I'd much rather resolve this by getting rid of the async channel handling